trading_calendars/exchange_calendar_xbue.py

```diff
@@ -0,0 +1,373 @@
+    - Day of Respect for Cultural Diversity (2nd Monday in Oct)
+    - Bank Holiday (Nov 6)
+    - Day of National Sovereignty (4th Monday in Nov)
+    - Immaculate Conception
+    - Christmas Eve (2005-present)
+    - Christmas Day
+    - Last Trading Day of Year (2005-present)
+
+
+    Early Closes:
+    Note - we are unsure if these are early closes or holidays
+    - Christmas Eve
+    - New Year's Eve
+    """
+    name = 'XBUE'
+
+    tz = timezone('America/Argentina/Buenos_Aires')
+
+    open_times = (
+        (None, time(11, 1)),
+    )
+
+    close_times = (
+        (None, time(17, 00)),
+    )
+
+    # TODO: Verify early close time
+    regular_early_close = time(13)
+
+    @property
+    def regular_holidays(self):
+        return HolidayCalendar([
+            NewYearsDay,
+            CarnivalMonday,
+            CarnivalTuesday,
+            TruthAndJusticeMemorialDay,
+            MalvinasDayTo2004,
+            MalvinasDayFrom2005,
+            MaundyThursday,
+            GoodFriday,
+            LabourDay,
+            MayDayRevolution,
+            MartinMiguelDeGuemesDay,
+            NationalFlagDayTo2010,
+            NationalFlagDayFrom2011,
+            IndependenceDay,
+            SanMartinsDayTo2010,
+            SanMartinsDayFrom2012,
+            DayOfRespectForCulturalDiversity,
+            BankHoliday,
+            DayOfNationalSovereigntyTo2016,
+            DayOfNationalSovereigntyFrom2017,
+            ImmaculateConception,
+            ChristmasDay,
+        ])
+
+    @property
+    def adhoc_holidays(self):
+        misc_adhocs = [
+            '2009-07-10',
+            '2010-05-24',
+            '2010-10-27',
+            '2012-01-02',
+            '2012-02-27',
+            '2012-09-24',
+            '2013-01-31',
+            '2013-02-20',
+            '2018-11-30',
+        ]
+
+        # There was a string of market closures in January, 2002
+        market_closures_2002_jan = [
+            '2002-01-{:02d}'.format(day) for day in range(7, 17)
+        ]
+
+        # There was a string of market closures in April, 2002
+        market_closures_2002_apr = [
+            '2002-04-{:02d}'.format(day) for day in range(22, 27)
+        ]
+
+        # There are occasionally "bridge days" that make long weekends when
+        # a holiday falls on a Tuesday/Thursday, but these days are chosen
+        # inconsistently.
+        bridge_days = [
+            '2011-03-25',
+            '2011-12-09',
+            '2012-04-30',
+            '2013-04-01',
+            '2013-06-21',
+            '2014-05-02',
+            '2014-12-26',
+            '2015-03-23',
+            '2015-12-07',
+            '2016-07-08',
+            '2016-12-09',
+            '2018-04-30',
+            '2018-12-24',
+            '2018-12-31',
+            '2019-07-08',
+            '2019-08-19',
+            '2019-10-14',
+        ]
+
+        irregular_observances = [
+            '2012-10-08',  # Cultural Diversity Day
```

Comment:
Does this mean it was observed twice this year? Because the regular holiday doesn't omit 2012.

Reply:
The regular holiday actually does omit 2012, in `cultural_diversity_observance` the last line is filtering out 2012, which is why I also had to manually map `nearest_monday` over the holidays rather than just feed it directly to the observance parameter (which I thought was ugly/annoying but I wasn't sure how else to do it).

Reply:
Ah missed that line 👍